comms/dht/src/actor.rs

```diff
@@ -835,8 +904,12 @@ impl DhtActor {
                 .chain(candidates.iter())
                 .cloned()
                 .collect::<Vec<_>>();
-            // If we don't have enough connections, let's select some more disconnected peers (at least 2)
-            let n = cmp::max(config.broadcast_factor.saturating_sub(candidates.len()), 2);
+            // If we don't have enough connections, let's select some more disconnected peers (at least 2, but account
+            // for connected seed peers)
+            let n = cmp::max(
+                config.broadcast_factor.saturating_sub(candidates.len()) + connected_seeds_len,
```

Comment:
why are we now including the seed peers here, are they not counted above?

Reply:
No, they are not counted. The `candidates` considers any current connections, then when we branch into `if !closest_request.connected_only {` we want to disregard/not acknowledge seed connections. As an example, if we want to broadcast to 8 peers, and we have 7 connections, but 5 of those connections are to seeds, we want to select 5 other peers to broadcast to in addition to the 1 we are missing. This is to ensure that we are not overly reliant on seed peers to have a well-connected network.